tests/basic/pure_test.go

```diff
@@ -4238,6 +4238,248 @@ var _ = Describe("{CreateNewPoolsWhenFadaFbdaVolumeCreationInProgress}", func()
+		scNamespaceMap := map[string]string{
+			baseScName: BaseAppNameSpace,
+			fadaScName: FadaAppNameSpace,
+			fbdaScName: FbdaAppNameSpace,
+		}
+
+		//Get The Details of Existing FA AND FB in the cluster
+		flashArrays, err := GetFADetailsUsed()
+		log.FailOnError(err, "Failed to get FA details from pure.json in the cluster")
+		flashBlades, err := GetFBDetailsFromCluster()
+		log.FailOnError(err, "Failed to get FB details from pure.json in the cluster")
+
+		stepLog := "Create storage class with max iops and max bandwidth for Normal Portworx Volumes , FADA and FBDA Pvc Deployment"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			BaseParams := make(map[string]string)
+			BaseParams["repl"] = "1"
+			BaseParams["max_iops"] = "1000"
+			BaseParams["max_bandwidth"] = "1G"
+			reclaimPolicyDelete := v1.PersistentVolumeReclaimDelete
+			bindMode := storageApi.VolumeBindingImmediate
+			// create storage class for base volumes
+			_, err := CreatePortworxStorageClass(baseScName, reclaimPolicyDelete, bindMode, BaseParams)
+			log.FailOnError(err, "Failed to create base storage class")
+			log.InfoD("Storage class [%s] for Basic is created", baseScName)
+
+			faParams := make(map[string]string)
+			faParams["repl"] = "1"
+			faParams["max_iops"] = "1000"
+			faParams["max_bandwidth"] = "1G"
+			faParams["fs"] = "ext4"
+
+			var allowVolExpansionFA bool = true
+			// create storage class for FADA volumes
+			err = CreateFlashStorageClass(fadaScName,
+				"pure_block",
+				v1.PersistentVolumeReclaimDelete,
+				faParams, nil, &allowVolExpansionFA,
+				storageApi.VolumeBindingImmediate,
+				nil)
+			log.FailOnError(err, fmt.Sprintf("Failed to create storage class [%v] ", fadaScName))
+			log.InfoD("Storage class [%s] for FADA is created", fadaScName)
+
+			fbParams := make(map[string]string)
+			fbParams["pure_export_rules"] = "*(rw)"
+			// create storage class for FADA volumes
+			err = CreateFlashStorageClass(fbdaScName,
+				"pure_file",
+				v1.PersistentVolumeReclaimDelete,
+				fbParams, nil, &allowVolExpansionFA,
+				storageApi.VolumeBindingImmediate,
+				nil)
+			log.FailOnError(err, fmt.Sprintf("Failed to create storage class [%v] ", fbdaScName))
+			log.InfoD("Storage class [%s] for FBDA is created", fbdaScName)
+		})
+
+		//CreatePVC will create a pvc with given name, storage class, size and namespace
+		createPVC := func(pvcName string, scName string, pvcSize string, ns string) (*v1.PersistentVolumeClaim, error) {
+			size, err := resource.ParseQuantity(pvcSize)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse pvc size : %s", pvcSize)
+			}
+			pvcClaimSpec := k8s.MakePVC(size, ns, pvcName, scName)
+			pvc, err := k8sCore.CreatePersistentVolumeClaim(pvcClaimSpec)
+			return pvc, err
+		}
+		//createAndAppendPVC will call createPVC with given name, storage class, size and namespace
+		createAndAppendPVC := func(appName string, scName string, namespace string, x int) {
```

Comment:
[nit] "createAndAppendPVC" is a bit confusing in my opinion, it sounds more like it's appending to a list somewhere. Can we name it something like "createPVCWithAppNameAndIndex"